### DISCUSSION TITLE
Is disk cache in Coil 2.0 still needs specific header response?

### DISCUSSION THREAD

**Discussion by wiryadev** [coil-kt/coil#1346] [Q&A] [answered]
I store my data to local database so the app still can load cached data when offile. But the image seems to be failed to load. I wonder if it is because the header has `Cache-Control` set to `max-age=0`, because if im not mistaken Coil 2.x use new disk caching mechanism.
If i have to set custom header, is [it](https://coil-kt.github.io/coil/recipes/#headers) still the recommended way of doing it in 2.x?

**Comment by FishHawk** [comment, marked as answer]
According to coil's 2.0.0 change log:
>     - `Cache-Control` and other cache headers are still supported - except `Vary` headers, as the cache only checks that the URLs match.
Yes, coil still recognize the `Cache-Control` field. I think it would be better to use `addHeader` to avoid dropping other http header fields.
